qualtran/bloqs/factoring/mod_add_test.py

```diff
@@ -31,3 +34,9 @@ def test_ctrl_mod_add_k():
     counts = bloq.bloq_counts()
     ((bloq, n),) = counts.items()
     assert n == 5
+
+
+@pytest.mark.parametrize('bitsize,p', [(1, 1), (2, 3), (5, 8)])
+def test_montgomery_mod_add_decomp(bitsize, p):
+    bloq = MontgomeryModAdd(bitsize=bitsize, p=p)
+    assert_valid_bloq_decomposition(bloq)
```

Comment:
Can we add a correctness test using tensor network simulation for a small case, till the time the classical simulation is not available?

Reply:
I'm not familiar with tensor network simulation. Is there a doc or example test I can take a look at?

Reply:
For an effective test, one would need to implement the tensor for the new Bloq. In principle, this should be derivable from the classical action. I suggest in this instance we take out a techdebt loan and don't block this implementation on better testing infrastructure. The dividends come from the fact that it will be easier to develop and test the classical simulation and tensor stuff with these examples in the library

Reply:
I think it's worth overriding the `add_my_tensors` for the new Bloq at least. We can leave a TODO to fix the `on_classical_vals` and `add_my_tensors` for Bloqs that appear in the decomposition to verify that the decomposition is consistent with the Bloq action. But currently the Bloq doesn't define it's in terms of tensor network simulation and I think it's worth adding that in this PR. The code should be fairly straightforward, something like

```python
    def add_my_tensors(
        self,
        tn: 'qtn.TensorNetwork',
        tag: Any,
        *,
        incoming: Dict[str, 'SoquetT'],
        outgoing: Dict[str, 'SoquetT'],
    ):
        import quimb.tensor as qtn

        N = 2**self.bitsize
        inds = (incoming['x'], incoming['y'], outgoing['x'], outgoing['y'])
        data = np.zeros((N, N) * 2, dtype=np.complex128)
        # TODO: Add a value-to-index method on dtype to make this easier.
        for x, y in itertools.product(range(N), range(N)):
            out = self.on_classical_vals(x=x, y=y)
            data[x, y, out['x'], out['y']] = 1
        tn.add(qtn.Tensor(data=data, inds=inds, tags=[self.short_name(), tag]))
```